presto-main/src/main/java/com/facebook/presto/sql/planner/RowExpressionInterpreter.java

```diff
@@ -781,6 +797,40 @@ private SpecialCallResult tryHandleCast(CallExpression callExpression, List<Obje
                 return changed(null);
             }
 
+            // Only nested casts to varchar types are simplified.
+            if (source instanceof CallExpression && isVarcharType(targetType)) {
```

Comment:
I'm wondering what kind of queries would trigger nested varchar cast? Is it possible to have more than 2 casts? Since we want to handle this, why not handle it generically as arbitrary number of varchar casts.

Reply:
It should already be generically handled, with recursion at https://github.com/prestodb/presto/pull/14408/files#diff-6cf4df8e0f9122042d6c5a7aab910b78R829. For example 3-level nested cast: https://github.com/prestodb/presto/pull/14408/files#diff-a6bca70e7bdcc8feb45820d3bd17a0b2R791 I can add more tests with deeper levels of casts if necessary.